server/src/main/java/org/opensearch/cluster/coordination/JoinHelper.java

```diff
@@ -407,12 +447,42 @@ public String executor() {
     }
 
     public void sendValidateJoinRequest(DiscoveryNode node, ClusterState state, ActionListener<TransportResponse.Empty> listener) {
-        transportService.sendRequest(
-            node,
-            VALIDATE_JOIN_ACTION_NAME,
-            new ValidateJoinRequest(state),
-            new ActionListenerResponseHandler<>(listener, i -> Empty.INSTANCE, ThreadPool.Names.GENERIC)
-        );
+        if (node.getVersion().before(Version.V_2_9_0)) {
```

Comment:
> In main also, shouldn't we have pushed 2.9.0 in the version check for backward compatibility?

@shwetathareja Yes, we need to update main to have the 2.9.0 version guard as soon as this backport is merged. It means that we break compatibility as soon as this is merged, since we test main against the latest version on the 2.x branch. This is the standard compatibility dance we have to do and it is a somewhat painful process. Here is a recent issue to discuss this pain point: #8070

Reply:
Draft PR to fix the version guard on main: #8185